modules/react/common/stories/Responsive.stories.mdx

```diff
@@ -0,0 +1,314 @@
+the maximum width of that element at `1000px`. What we can do with Container Based Responsive
+Styling is measure when that container's width reaches certain breakpoints
+`(i.e. 1440px, 1024px, 768px, 320px, 0)` and apply styles based on those breakpoints. If the
+container's width were to reach `768px`, we can apply styles to that breakpoint.
+
+```tsx
+<Box maxWidth={1000}>Hello World</Box>
+```
+
+In the interactive example below, when the dropdown's value is updated, the container size will
+change to match the size shown in the dropdown. This is just an example of how container-based
+styles work and will be explained more in depth below.
+
+<ExampleCodeBlock code={ResponsiveContainer} />
+
+### Applying Container-Based Responsive Styling
+
+For container-based responsive styling, we have utilities that will allow you to apply responsive
+styles at those breakpoints. There are four utilities/hooks:
+`useResizeObserver | useResponsiveContainerStyles`
+
+`useResizeObserver` - This is a hook that will provide the width of the container. This can be used
+to measure container size as well.
+
+`useResponsiveContainerStyles` - This is a hook that will allow you to create container-based
+responsive styles with objects (as you can see in the example below). This hook accepts three
+arguments with the third being optional (Style Objects, Container Width, Theme). In each style
+object, there are five sizes that it will accept: `zero, small, medium, large and extra large`.
+These give sizes represent the breakpoints that are at the bottom of this page. The sizes will act
+like `min-width`. For example, if you want to apply styles from `medium` and up, then you would
+write those styles under `m`.
+
+In the example below, we are importing these utilities:
+`useResizeObserver | useResponsiveContainerStyles`
+
+See example below:
+
+```tsx
+import {Flex, Box} from '@workday/canvas-kit-react/layout';
+import {useResizeObserver, useResponsiveContainerStyles} from '@workday/canvas-kit-react/common';
+
+const ref = React.useRef(null);
+const [width, setWidth] = React.useState(0);
+
+useResizeObserver({
+  ref: ref,
+  onResize: data => {
+    setWidth(data.width || 0);
+  },
+});
+
+const containerResponsiveStyles = useResponsiveContainerStyles(
+  {
+    flex: {
+      flexDirection: 'column',
+      padding: 'm',
+      depth: 1,
+      borderRadius: 'l',
+      zero: {
+        backgroundColor: 'Red',
+      },
+      s: {
+        backgroundColor: 'Orange',
+      },
+      m: {
+        backgroundColor: 'Yellow',
+      },
+      l: {
+        backgroundColor: 'Green',
+      },
+      xl: {
+        backgroundColor: 'Blue',
+      },
+    },
+    box: {
+      padding: 's',
+    },
+  },
+  width
+);
+
+return (
+  <Box ref={ref}>
+    <Flex {...containerResponsiveStyles.flex}>
+      <Box {...containerResponsiveStyles.box}>Hello World</Box>
+    </Flex>
+  </Box>
+);
+```
```

Comment:
Again, I don't think we need this example. It doesn't seem to add much to the interactive example above. Let me know if I'm missing something.

```suggestion
```